components/google_calendar/google_calendar.app.js

```diff
@@ -1,154 +1,307 @@
-      return calendar
+      const auth = new google.auth.OAuth2();
+      auth.setCredentials(this._tokens());
+      const calendar = google.calendar({
+        version: "v3",
+        auth,
+      });
+      return calendar;
     },
     async calendarList() {
-      const calendar = this.calendar()
-      const resp = await calendar.calendarList.list()
-      return resp
+      const calendar = this.calendar();
+      const resp = await calendar.calendarList.list();
+      return resp;
     },
     async list(config) {
-      const calendar = this.calendar()
-      const resp = await calendar.events.list(config)
-      return resp
+      const calendar = this.calendar();
+      const resp = await calendar.events.list(config);
+      return resp;
     },
     // for config key value pairs - https://developers.google.com/calendar/v3/reference/events/list
     // deprecated
     async getEvents(config) {
-      return await this.list(config)
+      return await this.list(config);
     },
     async watch(config) {
-      const calendar = this.calendar()
-      const resp = await calendar.events.watch(config)
-      return resp
+      const calendar = this.calendar();
+      const resp = await calendar.events.watch(config);
+      return resp;
     },
     async stop(config) {
-      const calendar = this.calendar()
-      const resp = await calendar.channels.stop(config)
-      return resp
+      const calendar = this.calendar();
+      const resp = await calendar.channels.stop(config);
+      return resp;
     },
     async fullSync(calendarId) {
-      let nextSyncToken = null
-      let nextPageToken = null
-      while(!nextSyncToken) {
+      let nextSyncToken = null;
+      let nextPageToken = null;
+      while (!nextSyncToken) {
         const listConfig = {
           calendarId,
-          pageToken: nextPageToken
-        }
-        const syncResp = await this.list(listConfig)
-        nextPageToken = get(syncResp, "data.nextPageToken")
-        nextSyncToken = get(syncResp, "data.nextSyncToken")
+          pageToken: nextPageToken,
+        };
+        const syncResp = await this.list(listConfig);
+        nextPageToken = get(syncResp, "data.nextPageToken");
+        nextSyncToken = get(syncResp, "data.nextSyncToken");
       }
-      return nextSyncToken
-    }
-  }
-}
+      return nextSyncToken;
+    },
+    /**
+     * @param {string} [pageToken] - the page token for the next page of shared
+     * calendars
+     * @returns
+     */
+    async listCalendarsInPage(pageToken) {
+      const calendar = this.calendar();
+      const { data } = await calendar.calendarList.list({
+        pageToken,
+        showHidden: true,
+      });
+      return data;
+    },
+
+    /**
+     * @param {string} [calendarId] - user calender id
+     * @returns
+     */
+    async listEventOptions(calendarId) {
+      const calendar = this.calendar();
+      const { data } = await calendar.events.list({
+        calendarId,
+      });
```

Comment:
here if you refactor the function `list` to `eventsList` the you can change this call to
```suggestion
      const data = await this.listEvents({
        calendarId,
      });
```